papas/detectors/Detector.h

```diff
@@ -48,17 +48,20 @@ class Detector {
   double muonAcceptance(const Track& track) const { return track.p3().Perp() > 5. && fabs(track.p3().Eta()) < 2.5; }
 
   double muonPtResolution(const PFParticle& /*ptc*/) const { return 0.02; }
-
+  
+  const std::list<std::shared_ptr<const DetectorElement>>  elements() const { return m_elements;}
+  
 protected:
+  ///sets up a list of all detector elements m_elements (needed for propagation)
+  void setElements();
```

Comment:
not clear what this method is doing. 
usually set methods are used to set an attribute, and thus take an argument. 
also, the documentation string is too vague.